src/nvim/api/vim.c

```diff
@@ -1245,6 +1245,179 @@ ArrayOf(Dictionary) nvim_get_keymap(String mode)
   return keymap_array(mode, NULL);
 }
 
+/// Sets a |mapping| for the given mode.
+///
+/// Note that isn't possible to specify a target buffer when opts contains the
+/// key "buffer"; like with |:map-<buffer>|, only the current buffer will be
+/// affected.
+///
+/// @param  mode  Mode short-name (the first character of an :[ ]map command,
+///               e.g. "n", "i", "v", "x", etc.) OR the exact string "m!" (for
+///               |:map!|). Unrecognized or empty values are treated like "m".
+/// @param  maptype   Whether to |:map|, |:unmap|, or |:noremap|, represented by
+///                   an empty string, "u", and "n", respectively.
+/// @param  opts  Optional parameters. Includes all |:map-arguments| as keys.
```

Comment:
typically `opts` is the last arg (but before `Error`). Urrecognized keys should be an error.